Adding first test to generate prototype

vsoch opened the pull request:
This includes the pull request action, and when that is working I'll use the base to generated the same example for automated.

Signed-off-by: Vanessa Sochat <vsochat@stanford.edu>

vsoch commented:
hey @Beanow I need your help here! I have the prototype pull request workflow (I think) figured out based on reading [documentation here](https://github.com/sourcecred/sourcecred#build), and I've now tested this both in the workflow and locally and am getting the same result - the "docs" folder I'm binding to the container is empty. Here are basic commands:

```bash
$ export GITHUB_REPOSITORY=sourcecred/sourcecred-action
$ SOURCECRED_GITHUB_TOKEN=${GITHUB_TOKEN} docker run -i --rm -v $PWD/sourcecred_data:/data -e SOURCECRED_GITHUB_TOKEN sourcecred/sourcecred:dev load ${GITHUB_REPOSITORY}
```
I list the `sourcecred_data` to confirm that it exists (it does). Then I want to run the build:

```bash
$  SOURCECRED_GITHUB_TOKEN=${GITHUB_TOKEN} docker run -i -d --rm -v $PWD/sourcecred_data:/data -e SOURCECRED_GITHUB_TOKEN -v $PWD/docs:/output sourcecred/sourcecred:dev build --output-path /output
```
And then docs is empty, both here in GitHub workflows and locally. Am I doing something wrong?

Beanow commented on .github/workflows/prototype_pull_request.yml:
```diff
@@ -0,0 +1,56 @@
+      run: |
+        mkdir -p sourcecred_data docs
+        SOURCECRED_GITHUB_TOKEN=${TOKEN} docker run -i --rm -v $PWD/sourcecred_data:/data -e SOURCECRED_GITHUB_TOKEN sourcecred/sourcecred:dev load ${GITHUB_REPOSITORY}
+        SOURCECRED_GITHUB_TOKEN=${TOKEN} docker run -i -d --rm -v $PWD/sourcecred_data:/data -e SOURCECRED_GITHUB_TOKEN -v $PWD/docs:/output sourcecred/sourcecred:dev build --output-path /output
```
The build step shouldn't need a github token, so that and the env flag can be removed. It will read from the `sourcecred_data` dir.

Also seeing a `-d` daemon flag. Perhaps that's the issue here?

vsoch replied:
Testing locally - removing the -d fixed the issue (and I'll push soon). It looks like the `-d` is shown in the example, so if we want folks to be able to watch progress of the generation, maybe we should remove it from there (more likely to hit unseen bugs having it?) For your interest, here is the output printed to the terminal:

```bash
Removing contents of build directory: /output
(node:17) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
PROJECT_IDS: ["sourcecred/sourcecred-action"]
Hash: 0bb2fd071b7be0132950
Version: webpack 4.41.2
Time: 31088ms
Built at: 12/28/2019 5:56:46 PM
                                            Asset       Size  Chunks                                Chunk Names
                              asset-manifest.json  712 bytes          [emitted]                     
                        discord-invite/index.html   6.23 KiB          [emitted]                     
                                      favicon.png   2.52 KiB          [emitted]                     
                                       index.html   12.2 KiB          [emitted]                     
                             prototype/index.html   6.34 KiB          [emitted]                     
prototype/sourcecred/sourcecred-action/index.html   6.82 KiB          [emitted]                     
                       static/js/main.4d0ea202.js    762 KiB       0  [emitted] [immutable]  [big]  main
                   static/js/main.4d0ea202.js.map   3.51 MiB       0  [emitted] [dev]               main
                        static/js/ssr.c5f36d9f.js    786 KiB       1  [emitted] [immutable]  [big]  ssr
                    static/js/ssr.c5f36d9f.js.map   3.56 MiB       1  [emitted] [dev]               ssr
                     test/FileUploader/index.html   7.22 KiB          [emitted]                     
                 test/TimelineCredView/index.html   22.2 KiB          [emitted]                     
 timeline/sourcecred/sourcecred-action/index.html    6.1 KiB          [emitted]                     
Entrypoint main [big] = static/js/main.4d0ea202.js static/js/main.4d0ea202.js.map
Entrypoint ssr [big] = static/js/ssr.c5f36d9f.js static/js/ssr.c5f36d9f.js.map
 [28] ./node_modules/react-router/es/index.js + 32 modules 76.4 KiB {0} {1} [built]
      |    33 modules
 [45] ./src/util/pathNormalize.js 13.4 KiB {0} {1} [built]
 [72] ./src/homepage/routeData.js 3.39 KiB {0} {1} [built]
[104] ./src/webutil/assets.js 1.98 KiB {0} {1} [built]
[105] ./src/util/dedent.js 1.03 KiB {0} {1} [built]
[142] ./src/homepage/ExternalRedirect.js 652 bytes {0} {1} [built]
[143] ./src/homepage/Page.js + 4 modules 8.43 KiB {0} {1} [built]
      | ./src/homepage/Page.js 3.02 KiB [built]
      | ./src/homepage/GithubLogo.js 1.21 KiB [built]
      | ./src/homepage/TwitterLogo.js 1.14 KiB [built]
      | ./src/homepage/DiscordLogo.js 1.21 KiB [built]
      | ./src/core/version.js 1.81 KiB [built]
[209] ./src/webutil/createRelativeHistory.js 7.79 KiB {0} {1} [built]
[222] ./src/homepage/createRouteDataFromEnvironment.js 242 bytes {0} {1} [built]
[223] ./src/homepage/createRoutes.js + 1 modules 2.31 KiB {0} {1} [built]
      | ./src/homepage/createRoutes.js 1.19 KiB [built]
      | ./src/webutil/withAssets.js 1.09 KiB [built]
[251] ./config/polyfills.js 853 bytes {0} {1} [built]
[545] multi ./config/polyfills.js ./src/homepage/index.js 40 bytes {0} [built]
[548] multi ./config/polyfills.js ./src/homepage/server.js 40 bytes {1} [built]
[549] ./src/homepage/server.js 5.61 KiB {1} [built]
[553] ./src/homepage/index.js + 1 modules 1.87 KiB {0} [built]
      | ./src/homepage/index.js 1.37 KiB [built]
      | ./src/homepage/App.js 468 bytes [built]
    + 726 hidden modules

WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
This can impact web performance.
Assets: 
  static/js/main.4d0ea202.js (762 KiB)
  static/js/ssr.c5f36d9f.js (786 KiB)

WARNING in entrypoint size limit: The following entrypoint(s) combined asset size exceeds the recommended limit (244 KiB). This can impact web performance.
Entrypoints:
  main (762 KiB)
      static/js/main.4d0ea202.js
  ssr (786 KiB)
      static/js/ssr.c5f36d9f.js


WARNING in webpack performance recommendations: 
You can limit the size of your bundles by using import() or require.ensure to lazy load some parts of your application.
For more info visit https://webpack.js.org/guides/code-splitting/
```

Beanow replied:
One line in particular from the output is looking good.
One of the first lines: `PROJECT_IDS: ["sourcecred/sourcecred-action"]`
This lists the projects our build step found in the `sourcecred_data`.

Beanow commented:
Oh I think there may be one part missing after the build step.
This copies part of the `sourcecred_data` to an `api/v1/data` dir.

https://github.com/teamopen-dev/shared-sourcecred/blob/ff6a170eb1b79fe00e2037573b8adb8aad31f2f5/scripts/rebuild-site.sh#L47-L51

vsoch commented:
Does that look better? I'm going to do one more tweak so that we can customize the Title, and then add the automated example, run it once more, and then remove the pull_request trigger entirely and merge.

vsoch commented:
okay, I've added the automated example, and I've removed the "pull_request" trigger so it won't run here. Importantly, the pull_request.sh script was updated, and with a sha not in master, so if we ran it from here it wouldn't be able to download the file. Since both the workflows will trigger early on Sunday, and the last few tests worked, I think our best bet is to merge, and then I'll look for the actions tomorrow early and do any remaining updates that are needed. Once they have run and we've generated an example prototype, then we can share with others again for more feedback. 

So if it's okay with you @Beanow I'm ready to merge!

Beanow commented:
Looking at the commands the copy seems right. But it's not listed in the logs for `git add`?

```
  13 files changed, 319 insertions(+)
 create mode 100644 docs/asset-manifest.json
 create mode 100644 docs/discord-invite/index.html
 create mode 100644 docs/favicon.png
 create mode 100644 docs/index.html
 create mode 100644 docs/prototype/index.html
 create mode 100644 docs/prototype/sourcecred/sourcecred-action/index.html
 create mode 100644 docs/static/js/main.e6ab1a42.js
 create mode 100644 docs/static/js/main.e6ab1a42.js.map
 create mode 100644 docs/static/js/ssr.253a84fe.js
 create mode 100644 docs/static/js/ssr.253a84fe.js.map
 create mode 100644 docs/test/FileUploader/index.html
 create mode 100644 docs/test/TimelineCredView/index.html
 create mode 100644 docs/timeline/sourcecred/sourcecred-action/index.html
```

Beanow commented:
Oh! I see the issue. The build command clears your output directory. So the copy should be done after build.

vsoch commented:
Ah okay, I'll fix that up. I thought those files were needed for the build, hence copying them first.

vsoch commented:
Look okay now?

Beanow commented:
>  I thought those files were needed for the build, hence copying them first.

Can't blame you :stuck_out_tongue: it's a bit like https://discourse.sourcecred.io/t/sourcecred-beta-roadmap/422/4 atm.

As the PR doesn't trigger builds now I'm going to :+1: based on scrolling through the script so we can run it from master.

vsoch commented:
Woot! I'll follow up tomorrow when it runs. Thanks for your help @Beanow !
